in_toto/verifylib.py

```diff
@@ -1115,105 +969,79 @@ def verify_item_rules(source_name, source_type, rules, links):
-
-    elif rule_type == "allow":
-      source_artifacts_queue = verify_allow_rule(rule, source_artifacts_queue)
-
-    elif rule_type == "disallow":
-      verify_disallow_rule(rule, source_artifacts_queue)
-
-
-    # CREATE, DELETE and MODIFY always operate either on products, on materials
-    # or both, independently of the source_type ...
-    elif rule_type == "create":
-      source_products_queue = verify_create_rule(
-          rule, source_materials_queue, source_products_queue)
-
-      # The create rule only updates the products_queue, which in turn
-      # only affects the generic artifacts queue if source_type is "products"
-      if source_type == "products":
-        source_artifacts_queue = source_products_queue
-
-    elif rule_type == "delete":
-      source_materials_queue = verify_delete_rule(
-          rule, source_materials_queue, source_products_queue)
-
-      # The delete rule only updates the materials_queue, which in turn
-      # only affects the generic artifacts queue if source_type is "materials"
-      if source_type == "materials":
-        source_artifacts_queue = source_materials_queue
-
-    # NOTE: Can't reach `else` branch, if the rule is none of these types
-    # an exception would have been raised above in `unpack_rule`
-    elif rule_type == "modify": # pragma: no branch
-      # The modify rule updates materials_queue and products_queue. We have to
-      # update the generic artifacts queue accordingly.
-      if source_type == "materials":
-        source_materials_queue, source_products_queue = verify_modify_rule(
-            rule, source_artifacts_queue, source_products_queue,
-            source_materials, source_products)
-        source_artifacts_queue = source_materials_queue
-
-      # NOTE: Can't reach `else` branch, if the source_type is none of these
-      # types an exception would have been raised above in `unpack_rule`
-      elif source_type == "products": # pragma: no branch
-        source_materials_queue, source_products_queue = verify_modify_rule(
-            rule, source_materials_queue, source_artifacts_queue,
-            source_materials, source_products)
-        source_artifacts_queue = source_products_queue
-
-    # Store all available queues of rule in results and add to traceback
-    results_dict["rule"] = rule
-    results_dict["artifacts"] = source_artifacts_queue
-    results_dict["materials"] = source_materials_queue
-    results_dict["products"] = source_products_queue
-    RULE_TRACE["trace"].append(results_dict)
+    elif _type == "allow":
+      consumed = verify_allow_rule(_pattern, artifacts_queue)
+
+    # It's up to the "disallow" rule to raise an error if artifacts were not
+    # consumed as intended
+    elif _type == "disallow":
+      verify_disallow_rule(_pattern, artifacts_queue)
+
+    else: # pragma: no cover (unreachable)
+      raise securesystemslib.exceptions.FormatError(
+          "Invaldid rule type '{}'.".format(_type))
```

Comment:
Now that we've coalesced this interfaces almost there, wouldn't it be worth moving this into something more akin to a dispatch table??

Reply:
Sure, why not, as long as it enhances readability. How would you go about different parameters? Use some `*args`/`**kwargs` wizardry? For comparison, [here's how we do it in the go implementation](https://github.com/in-toto/in-toto-golang/blob/ef3bff026632eaf918f5f964dcb894a136a6260c/in_toto/verifylib.go#L257-L285).